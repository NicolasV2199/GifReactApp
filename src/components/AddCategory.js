import React, { useState } from 'react';
import PropTypes from 'prop-types';



//Recibo por parametros la funcion de set categories para agregar nuevas
export const AddCategory = ({ setCategories }) => {

    //Uso el hook useState para poder cambiar el valor del input
    const [inputValue, setInputValue] = useState('');

    //Me encargo de que cuando haya un cambio en el input, se muestre el nuevo valor
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    /* //Cuando se presione la tecla Enter prevengo que se refresque la pagina, y hago la validacion de que si lo que se inserta es una categoria, entonces llamo a la funcion setCategories que me retorna las categorias que ya tenia, mas el valor insertado en el input, y reinicio el input a vacio*/
    const handleSubmit = (e) =>{
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories( cat => [ inputValue, ...cat,] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input 
                type="text" 
                value={ inputValue } 
                onChange={ handleInputChange } 
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
