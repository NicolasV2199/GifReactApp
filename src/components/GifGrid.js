import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs( category );

    //Se utiliza para que la funcion getgifs solo se llame cuando se renderiza el componente por primera vez
    /* Tiene un parametro que es un arreglo de dependencias, en este caso, vacio */
    //Se agrega la categoria en el arreglo de dependencias por si acaso llegase a cambiar.

    return (
        <>
            <h2 className='animate__animated animate__fadeInDown'> {category} </h2>

            {loading && <p className='animate__animated animate__flash animate__slow'>Loading...</p>}

            <div className='card-grid'>
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
