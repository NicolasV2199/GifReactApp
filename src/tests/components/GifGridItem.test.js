import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';




describe('Pruebas en el GifGridItem', () => {
  
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } />);

    test('Debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el titulo', () => {
      
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title ); 
    });

    test('Debe obtener la imagen igual al url y alt de los props', () => {
      
        const img = wrapper.find('img');

        expect( url ).toBe( img.prop( 'src' ) );
        expect( title ).toBe( img.prop( 'alt' ) );
    });

    test('Debe tener la clase animate__backInLeft', () => {
      
        const div = wrapper.find('div');
        const animation = 'animate__backInLeft';

        expect( div.prop('className')).toContain ( animation );
    });
    
    
    
    
});
