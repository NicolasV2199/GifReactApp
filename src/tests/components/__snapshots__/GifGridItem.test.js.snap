// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
